serialization/json_value.go

```diff
@@ -45,3 +49,15 @@ func (j *JSON) UnmarshalJSON(data []byte) error {
 	*j = append((*j)[0:0], data...)
 	return nil
 }
+
+// Scan is implemented to satisfy sql.Scanner interface, to scan values.
+func (j *JSON) Scan(src interface{}) error {
```

Comment:
Do we need this function?

Reply:
Unfortunately yes, sql implementation refuses to scan custom types without this. Sharing a related resource just because I find the title amusing [Why We Can't Have Nice Things
](https://husobee.github.io/golang/database/2015/06/12/scanner-valuer.html) 😂

Reply:
`TestSQLQueryWithJSONValue` passes if I remove `Scan` from `serialization.JSON`. Could you check whether that is really required and the test is correct? Note that we don't implement that interface on `types.Decimal` and SQL tests involving that types also pass OK.

Reply:
You are right, thanks for catching! Refactored deserialization at https://github.com/hazelcast/hazelcast-go-client/pull/726/commits/cb210c6750c45c535c0048fc613ef247cc287e31